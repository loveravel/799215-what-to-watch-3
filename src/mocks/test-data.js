export const films = [
  {
    id: 1,
    name: `Movie 1`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: true
  },
  {
    id: 2,
    name: `Movie 2`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideoLink: ``,
    video: ``,
    isFavorite: false
  },
  {
    id: 3,
    name: `Movie 3`,
    genre: `Drama`,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: false
  },
];

export const movie = {
  id: 1,
  name: `Movie 1`,
  genre: ``,
  released: 2019,
  backgroundImage: ``,
  backgroundColor: ``,
  previewImage: ``,
  posterImage: ``,
  description: ``,
  rating: 4.5,
  scoresCount: 300,
  runTime: 3,
  director: ``,
  starring: [],
  previewVideo: ``,
  video: ``,
  isFavorite: false
};

export const reviews = [
  {
    id: 1,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  }
];

export const userData = {
  id: 1,
  name: ``,
  email: ``,
  avatarUrl: ``
};
