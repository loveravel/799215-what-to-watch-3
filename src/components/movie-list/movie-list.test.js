import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import configureStore from "redux-mock-store";

import {films} from "../../mocks/test-data.js";
import history from "../../history.js";
import NameSpace from "../../reducer/name-space.js";

import MovieList from "./movie-list.jsx";

const mockStore = configureStore([]);

it(`<MovieList /> component renders correctly`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      films
    }
  });

  const props = {
    films,
    message: ``,
  };

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <MovieList {...props}/>
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`<MovieList /> component renders correctly without films`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      films: [],
    }
  });

  const props = {
    films: [],
    message: ``,
  };

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <MovieList {...props}/>
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();
  expect(tree).toMatchSnapshot();
});
