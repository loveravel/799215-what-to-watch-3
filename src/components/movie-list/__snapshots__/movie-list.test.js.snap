// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieList /> is rendered correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Aviator"
        height="175"
        src="img/aviator.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="aviator.html"
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Aviator"
        height="175"
        src="img/aviator.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="aviator.html"
      >
        Aviator
      </a>
    </h3>
  </article>
</div>
`;
